helpers/github.ts

```diff
@@ -347,18 +362,232 @@ export async function writeTotalRewardsToGithub(statistics: Statistics) {
+        labels: metaChanges.labels ? labelRemoved : originals,
+      });
+    } catch (err) {
+      console.error(err);
+    }
+
+    if (metaChanges.title || metaChanges.body || metaChanges.labels) console.log(`Updated metadata: ${devpoolIssue.html_url} (${projectIssue.html_url})`);
+  }
+
+  const hasNoPriceLabels = !(projectIssue.labels as GitHubLabel[]).some((label) => label.name.includes(LABELS.PRICE));
+
+  // these changes will open/close issues
+  const stateChanges: StateChanges = {
+    // missing in the partners
+    forceMissing_Close: {
+      cause: !projectIssues.some((projectIssue) => projectIssue.node_id == getIssueLabelValue(devpoolIssue, "id:")),
+      effect: "closed",
+      comment: "Closed (missing in partners)",
+    },
+    // no price labels set and open in the devpool
+    noPriceLabels_Close: {
+      cause: hasNoPriceLabels && devpoolIssue.state == "open",
+      effect: "closed",
+      comment: "Closed (no price labels)",
+    },
+    // it's closed, been merged and still open in the devpool
+    issueComplete_Close: {
+      cause: projectIssue.state == "closed" && devpoolIssue.state == "open" && !!projectIssue.pull_request?.merged_at,
+      effect: "closed",
+      comment: "Closed (merged)",
+    },
+    // it's closed, assigned and still open in the devpool
+    issueAssignedClosed_Close: {
+      cause: projectIssue.state == "closed" && devpoolIssue.state == "open" && !!projectIssue.assignee?.login,
+      effect: "closed",
+      comment: "Closed (assigned-closed)",
+    },
+    // it's closed, not merged and still open in the devpool
+    issueClosed_Close: {
+      cause: projectIssue.state == "closed" && devpoolIssue.state == "open",
+      effect: "closed",
+      comment: "Closed (not merged)",
+    },
+
+    // it's open, assigned and still open in the devpool
+    issueAssignedOpen_Close: {
+      cause: projectIssue.state == "open" && devpoolIssue.state == "open" && !!projectIssue.assignee?.login,
+      effect: "closed",
+      comment: "Closed (assigned-open)",
+    },
+    // it's open, merged, unassigned and closed in the devpool
+    issueReopenedMerged_Open: {
+      cause:
+        projectIssue.state == "open" &&
+        devpoolIssue.state == "closed" &&
+        !!projectIssue.pull_request?.merged_at &&
+        !hasNoPriceLabels &&
+        !projectIssue.assignee?.login,
+      effect: "open",
+      comment: "Reopened (merged)",
+    },
+    // it's open, unassigned and closed in the devpool
+    issueUnassigned_Open: {
+      cause: projectIssue.state == "open" && devpoolIssue.state == "closed" && !projectIssue.assignee?.login && !hasNoPriceLabels,
+      effect: "open",
+      comment: "Reopened (unassigned)",
+    },
+  };
+
+  let newState: "open" | "closed" | undefined = undefined;
+
+  // then process the state changes
+  for (const [, value] of Object.entries(stateChanges)) {
```

Comment:
Why not `Object.values` if the key is not used?

Reply:
for of guarantees sequential execution as far as I understand which is preferred here

Reply:
Not sure to understand why `entries` would keep it and not `values`.
```JavaScript
const obj = {
  foo: "bar",
  bar: "baz",
  baz: "boz",
};

for (const [, value] of Object.entries(obj)) {
  console.log(value);
}

console.log("-------");

for (const value of Object.values(obj)) {
  console.log(value);
}
```
would output
```text
bar
baz
boz
-------
bar
baz
boz
```
Objects are somehow predictable since ES2015 (see https://stackoverflow.com/questions/5525795/does-javascript-guarantee-object-property-order/38218582#38218582) but not guaranteed. So if that is a concern it should be changed to use an array or a map.

Reply:
my mistake, I thought you meant to use

```js
Object.values().map // or .forEach()
```